src/Microsoft.DotNet.Build.CloudTestTasks/CreateAzureContainer.cs

```diff
@@ -1,112 +1,159 @@
+        ///     When false, if the specified container already exists get a reference to it.
+        ///     When true, if the specified container already exists the task will fail.
         /// </summary>
         public bool FailIfExists { get; set; }
 
+        internal bool FailState { get; set; }
+
+        static public byte[] bytestoWrite;
+        static public int bytesToWriteLength;
+        static public ManualResetEvent allDone = new ManualResetEvent(false);
+        static public string globalResponse;
+        static public HttpStatusCode globalStatusCode;
+
         public override bool Execute()
         {
-            Log.LogMessage(MessageImportance.High, "Creating container named '{0}' in storage account {1}.", ContainerName, AccountName);
-
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(string.Format("DefaultEndpointsProtocol=https;AccountName={0};AccountKey={1}", AccountName, AccountKey));
-            CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
-            CloudBlobContainer newContainer = blobClient.GetContainerReference(ContainerName);
+            Log.LogMessage(MessageImportance.High, "Creating container named '{0}' in storage account {1}.",
+                ContainerName, AccountName);
+            var dt = DateTime.UtcNow;
+            var url = string.Format("https://{0}.blob.core.windows.net/{1}?restype=container", AccountName, ContainerName);
+            StorageUri = String.Format("https://{0}.blob.core.windows.net/{1}/", AccountName, ContainerName);
+            var request = (HttpWebRequest) WebRequest.Create(url);
+            request.Method = "PUT";
+            request.Headers["x-ms-date"] = dt.ToString("R", CultureInfo.InvariantCulture);
+            request.Headers["x-ms-version"] =  AzureHelper.STORAGE_API_VERSION;
+            request.Headers["Authorization"] = AzureHelper.AuthorizationHeader(AccountName, AccountKey, "PUT", dt, request);
 
-            if (FailIfExists && newContainer.Exists())
+            if (ReadOnlyTokenDaysValid > 0)
             {
-                Log.LogError("The container '{0}' already exists.", ContainerName);
-                return false;
+                ReadOnlyToken = AzureHelper.createContainerSASToken(AccountName, ContainerName, AccountKey, "read", ReadOnlyTokenDaysValid);
             }
-            else
+
+            if (WriteOnlyTokenDaysValid > 0)
             {
-                newContainer.CreateIfNotExists();
+                WriteOnlyToken = AzureHelper.createContainerSASToken(AccountName, ContainerName, AccountKey, "write", WriteOnlyTokenDaysValid);
             }
 
-            StorageUri = newContainer.Uri.ToString();
+            Log.LogMessage(MessageImportance.High, "Sending request to create Container");
+            bytestoWrite = new byte[0];
+            bytesToWriteLength = 0;
+            request.BeginGetRequestStream(GetRequestStreamCallback, request);
```

Comment:
I strongly suggest using the GetRequestStreamAsync pattern here, it's much easier to use and read than the BeginFoo versions.  I'd also suggest "HttpClient", it's got a nicer interface, if you don't need to do anything complicated.